client/src/components/my-mentor-card/index.tsx

```diff
@@ -54,18 +53,13 @@ export default function MyMentorCard(props: {
   accessToken: string;
   continueAction: () => void;
 }): JSX.Element {
-  const { mentorState, editMentor } = useActiveMentor();
+  const { mentorState, editMentor, saveThumbnail } = useActiveMentor();
```

Comment:
don't do this. Do this:

```typescript
const thumbnail = useSelector<State, string>(s => {
    // selectActiveMentor can be a helper function exported from useActiveMentor or similar
   selectActiveMentor(s)?.thumbnail || " 
})
```

The reason has to do with scope of selectedState and how that impacts the triggering of render calls